core-service/src/main/java/com/crmfoodestablishment/coreservice/service/MenuService.java

```diff
@@ -14,36 +16,49 @@ public class MenuService {
 
     private final MenuRepository menuRepository;
 
+    private final MenuMapper menuMapper;
+
     private Menu checkIfMenuExist(Integer id) {
         return menuRepository.findById(id)
                 .orElseThrow(() -> new MenuNotFoundException("No menu found for this id"));
+
     }
 
-    public Menu addMenu(Menu menu) {
-        boolean idExist = menuRepository.existsById(menu.getId());
+    public Integer addMenu(MenuDto menuDto) {
+        boolean idExist = menuRepository.existsById(menuDto.getId());
         if (idExist) {
-            throw new IllegalStateException("Menu name: " + menu.getName() + " has been taken");
+            throw new IllegalStateException("Menu name: " + menuDto.getName() + " has been taken");
         }
-        return menuRepository.save(menu);
+        Menu menuToSave = menuMapper.fromMenuDtoToMenuEntity(menuDto);
+        Menu savedMenu = menuRepository.save(menuToSave);
+
+        MenuDto savedMenuDto = menuMapper.fromMenuEntityToMenuDto(savedMenu);
+        return savedMenuDto.getId();
     }
 
-    public List<Menu> findAllMenu() {
-        return menuRepository.findAll();
+    public List<MenuDto> findAllMenu() {
+        return menuRepository.findAll().stream()
+                .map(menuMapper::fromMenuEntityToMenuDto)
+                .toList();
     }
 
-    public Menu findByIdMenu(Integer id) {
-        return checkIfMenuExist(id);
+    public MenuDto findByMenuId(Integer id) {
```

Comment:
Consider changing name to findMenuById

Reply:
In the comments above, Polina write that it would be better to name the method findByMenuId. In the end, I don't really care), I can call it that if everyone is in favour.